src/dispatcher/server.rs

```diff
@@ -81,52 +77,67 @@ impl SignatureDispatcher for Ed25519SignatureDispatcher {
 
         let signatures = future::join_all(
             // map of Futures<Output=Result<SignWithKeyResponse, Error>>
-            confirmed_signers.clone().map(|signer|
-                async move {
-                    let channel = match self.connect_signer_tls(signer.endpoint.clone()).await {
-                        Ok(channel) => channel,
-                        Err(e) => {
-                            error!("Error connecting to Signer!");
-                            error!("Signer: {:?}", signer);
-                            error!("Error: {:?}", e);
-                            return Err(Status::internal(format!("Could not connect to the Signer `{}`, {}", signer.endpoint, e)))
-                        }
-                    };
-
-                    let mut client = SignerClient::new(channel);
-
-                    let req = tonic::Request::new(SignWithKeyRequest {
-                        pub_key: signer.pubkey.to_owned(),
-                        ms_essence: r.ms_essence.to_owned()
-                    });
-
-                    debug!("Sending request to Signer `{}`: {:?}", signer.endpoint, req);
-                    let res = client.sign_with_key(req).await;
-                    if res.is_err() {
-                        error!("Error getting response from Signer `{}`: {:?}", signer.endpoint, res);
+            confirmed_signers.clone().map(|signer| async move {
+                let channel = match self.connect_signer(signer.endpoint.clone()).await {
+                    Ok(channel) => channel,
+                    Err(e) => {
+                        error!("Error connecting to Signer!");
+                        error!("Signer: {:?}", signer);
+                        error!("Error: {:?}", e);
+                        return Err(Status::internal(format!(
+                            "Could not connect to the Signer `{}`, {}",
+                            signer.endpoint, e
+                        )));
                     }
```

Comment:
```suggestion
                    },
```
Aren't we doing this every time now?

Reply:
rust fmt didn't do it automatically

in my opinion if rustfmt doesn't do it automatically we shouldn't bother

Reply:
I agree